### ISSUE TITLE
Input group focus rings are truncated

### ISSUE THREAD

**Issue by garrett** [patternfly/patternfly#3926]
Input groups join widgets right next to each other. In the default state, that's fine. However, when one of the widgets in the group gets focus, the ring is chopped off for everything but the last child.

Screenshots from https://www.patternfly.org/v4/components/input-group (with focus):

![Screenshot from 2021-03-15 13-13-14](https://user-images.githubusercontent.com/10246/111152301-d9c0ad00-8590-11eb-9fd6-b71a11f39aff.png)

![Screenshot from 2021-03-15 13-12-53](https://user-images.githubusercontent.com/10246/111152303-da594380-8590-11eb-9b44-0aea4f085632.png)

![Screenshot from 2021-03-15 13-12-58](https://user-images.githubusercontent.com/10246/111152305-da594380-8590-11eb-9dc7-d72ea7c7350c.png)

![Screenshot from 2021-03-15 13-13-01](https://user-images.githubusercontent.com/10246/111152306-daf1da00-8590-11eb-8b89-15c235f82405.png)

**Comment by garrett** [comment]
Here's a simple CSS fix:

```css
.pf-c-input-group :focus {
  position: relative;
  z-index: 2;
}
```

As z-index is relative to its parent, this adds position to enable z-index (as a default static position doesn't work) and then makes it higher than its siblings. Since the overlapping part of the widget is drawn above with this little CSS tweak, the focus ring is drawn where expected.

**Comment by garrett** [comment]
Hmm... the position relative messes up dropdowns in the group. Going to investigate further.

**Comment by garrett** [comment]
Just using `z-index: 2` without the positioning works, at least in Firefox (while manipulating the DOM for the focused element). I've updated the above comment with the change.
